src/profile-logic/import/instruments/index.js

```diff
@@ -0,0 +1,910 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+
+import {
+  getEmptyProfile,
+  getEmptyThread,
+} from '../../../profile-logic/data-structures';
+import BinaryPlistParser, { UID } from './BinaryPlistParser';
+import BinReader from './BinReader';
+import MaybeCompressedReader from './MaybeCompressedReader';
+
+//types
+import type {
+  Profile,
+  FuncTable,
+  FrameTable,
+  StackTable,
+} from '../../../types/profile';
+import type { UniqueStringArray } from '../../../../src/utils/unique-string-array';
+
+type InstrumentsSample = {|
+  +timestamp: number,
+  +threadID: number,
+  +backtraceID: number,
+  backtraceStack?: Array<number>,
+|};
+
+type TraceDirectoryTree = {|
+  +name: string,
+  +files: Map<string, File>,
+  +subdirectories: Map<string, TraceDirectoryTree>,
+|};
+
+type InstrumentsFrameInfo = {|
+  +key: string | number,
+  +name: string,
+  file?: string,
+  +line?: number,
+  +col?: number,
+|};
+
+type FormTemplateRunData = {|
+  +number: number,
+  +addressToFrameMap: Map<number, InstrumentsFrameInfo>,
+|};
+
+function readAsArrayBuffer(file: File): Promise<ArrayBuffer> {
+  return MaybeCompressedReader.fromFile(file).readAsArrayBuffer();
+}
+
+function readAsText(file: File): Promise<string> {
+  return MaybeCompressedReader.fromFile(file).readAsText();
+}
+
+function parseBinaryPlist(bytes) {
```

Comment:
please add flow types: especially, what is `bytes` ?

Reply:
Added the type.